### PR TITLE
https://issues.redhat.com/browse/ACM-13416 Unknown state troubleshoot

### PR DESCRIPTION
**Pull request by oafischer**
2.11 only

### CODE REVIEW

**Comment by elgnay** [comment]
@oafischer In https://issues.redhat.com/browse/ACM-13416, two troubleshooting doc are proposed for separated scenarios although they are caused by the same issue and share the similar workaround:
- the first one happens during the installation of MCE;
- and the second one happens after the installation is completed;

In my opinion, it's better to separate those two scenarios due to the differing symptoms.

**Comment by oafischer** [comment]
@elgnay I will update it

**Comment by elgnay** [comment]
Please refer to https://github.com/stolostron/rhacm-docs/pull/6936 and use the correct name for ROSA HCP.

File: clusters/support_troubleshooting/trouble_install_rosa_hcp.adoc [hunk 1]
```diff
@@ -0,0 +1,78 @@
+[#trouble-install-stuck-rosa-hcp]
+= Troubleshooting installation status stuck at `Installing` on a {rosa} hosted cluster
+
+When installing {mce} on a {rosa} hosted cluster, {mce-short} becomes stuck at the `Installing` step and the `local-cluster` remains in the `Unknown` state.
+
+[#symptom-install-stuck-rosa-hcp]
+== Symptom: Installation status stuck at `Installing` on a {rosa} hosted cluster
+
+10 minutes after installing {mce-short}, the `local-cluster` remains in the `Unknown` state. When you check the `klusterlet-agent` pod log in the `open-cluster-management-agent` namespace on your managed cluster, you see an error that resembles the following:
```
Comment (elgnay) [inline clusters/support_troubleshooting/trouble_install_rosa_hcp.adoc]: on your managed cluster -> on the hub cluster

File: clusters/support_troubleshooting/trouble_install_rosa_hcp.adoc [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+----
+apiVersion: config.open-cluster-management.io/v1alpha1
+kind: KlusterletConfig
+metadata:
+  name: isrg-root-x1-ca
+spec:
+  hubKubeAPIServerCABundle: "<your_ca_certificate>"
+----
+
+. Apply the resource by running the following command:
```
Comment (elgnay) [inline clusters/support_troubleshooting/trouble_install_rosa_hcp.adoc]: Apply the resource by running the following command on the hub cluster:

**Comment by elgnay** [comment]
@oafischer the 2.11 workaround is different from the 2.10 workaround. It use the `global` klusterletconfig to simplify the procedure. Please refer to https://issues.redhat.com/browse/ACM-13416 for the details.

**Comment by oafischer** [comment]
Thanks for the review @elgnay. This PR isn't updated yet as I'm waiting to get the 2.10 version approved so that I can transfer the style and applicable changes.

**Comment by dockerymick** [comment]
@oafischer Please also update this PR similar to the suggestions that were provided for the 2.10 PR: https://github.com/stolostron/rhacm-docs/pull/6936

The {rosa} conref does exist for version 2.11: https://github.com/stolostron/rhacm-docs/blob/2.11_stage/modules/common-attributes.adoc

**Comment by oafischer** [comment]
I've updated the PR, please review again when you have time @elgnay @dockerymick. Thanks!

File: clusters/support_troubleshooting/troubleshooting_mce_intro.adoc [hunk 3]
```diff
@@ -42,4 +42,5 @@ To view the main documentation about managing your clusters, see xref:../cluster
  * xref:../support_troubleshooting/trouble_no_placementdecision_mce.adoc#troubleshooting-no-placementdecision-mce[Troubleshooting missing PlacementDecision after creating Placement]
  * xref:../support_troubleshooting/trouble_idrac_discovery_fails_mce.adoc#troubleshooting-idrac-discovery-fails-mce[Troubleshooting a discovery failure of bare metal hosts on Dell hardware]
  * xref:../support_troubleshooting/trouble_min_iso_boot.adoc#trouble-min-iso-boot[Troubleshooting Minimal ISO boot failures]
- 
+ * xref:../support_troubleshooting/trouble_install_rosa_hcp.adoc#trouble-install-stuck-rosa-hcp[Troubleshooting installation status stuck at `Installing` on a ROSA with hosted control planes cluster]
+ * xref:../support_troubleshooting/trouble_unknown_rosa_hcp.adoc#trouble-state-unknown-rosa-hcp[Troubleshooting all managed clusters become `Unknown` on a ROSA with hosted control planes cluster]
```
Comment (dockerymick) [inline clusters/support_troubleshooting/troubleshooting_mce_intro.adoc:46]: The link to the MCE RBAC is broken, can you help fix that?

**Review by dockerymick — APPROVED** [review]
One broken link. Otherwise, this looks good to me!

File: clusters/support_troubleshooting/troubleshooting_mce_intro.adoc (hunk 3 above)
Comment (oafischer) [reply clusters/support_troubleshooting/troubleshooting_mce_intro.adoc:46]: Should be fixed now

Comment (elgnay) [reply clusters/support_troubleshooting/troubleshooting_mce_intro.adoc:46]: @oafischer The 2.11 workaround utilizes the global KlusterletConfig to simplify the workaround provided in 2.10. However, the PR for 2.11 still includes the 2.10 workaround. Could you please update it based on the draft in https://issues.redhat.com/browse/ACM-13416.

Comment (oafischer) [reply clusters/support_troubleshooting/troubleshooting_mce_intro.adoc:46]: Updated

File: clusters/support_troubleshooting/trouble_install_rosa_hcp.adoc [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+oc apply -f <filename>
+----
+
+. Annotate your managed cluster by running the following command on the hub cluster:
+
++
+[source,bash]
+----
+oc annotate --overwrite managedcluster local-cluster agent.open-cluster-management.io/klusterlet-config='isrg-root-x1-ca'
+----
```
Comment (elgnay) [inline clusters/support_troubleshooting/trouble_install_rosa_hcp.adoc]: The annotation is no longer needed. The global KlusterletConfig will be adopted by all managed cluster by default.

File: clusters/support_troubleshooting/trouble_unknown_rosa_hcp.adoc [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+
+. Create a `KlusterletConfig` resource and add your encoded CA certificate to the `spec` parameter section. See the following example:
+
++
+[source,yaml]
+----
+apiVersion: config.open-cluster-management.io/v1alpha1
+kind: KlusterletConfig
+metadata:
+  name: isrg-root-x1-ca
```
Comment (elgnay) [inline clusters/support_troubleshooting/trouble_unknown_rosa_hcp.adoc]: `isrg-root-x1-ca` -> `global`

Comment (oafischer) [reply clusters/support_troubleshooting/trouble_unknown_rosa_hcp.adoc]: Updated, let me know if it looks good now @elgnay. Thanks!

**Comment by elgnay** [comment]
LGTM

**Pull request merged**